source/fundamentals/logging.txt

```diff
@@ -0,0 +1,370 @@
+
+.. tip:: Write Logs to a Specific File
+   
+   By default, the standard logger emits messages to your console (``stderr``). You can
+   specify a logging destination by setting the ``MONGODB_LOG_PATH``
+   environment variable to a filepath or ``stdout``.
+
+.. _golang-logging-component-severity:
+
+Logging Components and Severity Levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can specify which components the driver will log against by
+specifying the ``LogComponent`` type. The driver provides the following
+built-in specifications for ``LogComponent``:
+
+- ``LogComponentAll``: enables logging for all components.
+- ``LogComponentCommand``: enables command monitor logging.
+- ``LogComponentTopology``: enables topology logging.
+- ``LogComponentServerSelection``: enables server selection logging.
+- ``LogComponentConnection``: enables connection services logging.
+
+You can also specify the logging component by using the enumeration of
+the preceding specifications. The following code shows equivalent ways
+of enabling command monitoring:
+
+.. code-block:: go
+
+   // Using named value
+   comp := options.LogComponentCommand
+
+   // Using enumeration
+   comp := options.LogComponent(1)
+
+You can specify the log severity level by specifying the ``LogLevel``
+type. The driver provides the following
+built-in specifications for ``LogLevel``:
+
+- ``LevelOff``: suppresses logging.
+- ``LevelInfo``: enables logging of high-level informational messages.
+- ``LevelDebug``: enables logging of detailed debug messages.
+
+You can also specify the severity level by using the enumeration of
+the preceding specifications. The following code shows equivalent ways
+of enabling logging at the ``INFO`` level:
+
+.. code-block:: go
+
+   // Using named value
+   lvl := options.LogLevelInfo
+
+   // Using enumeration
+   lvl := options.LogLevel(1)
+
+.. important::
+
+   The {+driver-short+} currently supports only ``DEBUG`` level messages.
+
+To learn more about logging severity levels, see the Wikipedia entry on
+the :wikipedia:`Syslog standard for message logging <Syslog#Severity_level>`.
+
+Example
+~~~~~~~
+
+This example shows how to configure a basic logger that uses the standard
+logging library and prints log messages. The logger has the following
+specifications:
+
+- The maximum document length is ``25``.
+- The logging component is ``LogComponentCommand``.
+- The logging severity level is ``LogLevelDebug``.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/logging.go
+   :language: go
+   :dedent:
+   :emphasize-lines: 1-4, 9
+   :start-after: start-standard-logger
+   :end-before: end-standard-logger
+
+The following code performs an insert operation, which generates log
+messages:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/fundamentals/code-snippets/logging.go
+      :language: go
+      :dedent:
+      :start-after: start-insert
+      :end-before: end-insert
+
+   .. output::
```

Comment:
Q: Is this output what the user would see in the log?

Reply:
yes